hometask - 11

Leksa2304 opened the pull request:
Базовый уровень.
При верном нажатии буква окрашивается в зеленый, при неверном - в красный

Leksa2304 committed:
hometask - 11

Базовый уровень.
При верном нажатии буква окрашивается в зеленый, при неверном - в красный

Leksa2304 committed:
hometask-11

добавлен таймер, счет ошибок , правильных и неправильных слов.

работает только для одного слова.

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,114 @@
+
+function paintLetter(event) { //  функция окрашивания букв
+
+    if (event.key === "Shift" || event.key === "Control" || event.key === "CapsLock") return; // не принимать за ошибку нажатие смены языка и капслок
```
Бесполезный код. Удали плз

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,114 @@
+    wordContent.innerHTML = el.split("").map((char) => `<span>${char}</span>`).join("");
+}
+
+const spans = wordContent.querySelectorAll("span"); // spanы для каждой буквы
```
Лена,  файл прочитывается при перезагрузке страницы и все переменные прочитаются,  обработчики навесятся и сделаются выборки.  При нажатии на кнопки,  на клавиши,  вот этот код, написанные НЕ внутри обработчиков уже заново прочитан не будет. Т.е этот массив всегда будет неизменным и будет ориентироваться на первое слово.  В нем будет столько спанов, сколько букв в первом слове.

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,114 @@
+];
+
+function getword() { // функция генерации случайного слова
+    index = Math.floor(Math.random() * words.length);
```
Лена,  а почему без const?

Leksa2304 committed:
hometask-11

доработаны ошибки по окрашиванию слова за словом, реализован подсчет ошибок и окончание игры.

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+        "радость"
+    ];
+
+    index = Math.floor(Math.random() * words.length);
```
Если это переменная,  то почему без const?

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+let i = 0;
+let timerId;
+// let seconds = 0;
+// let minutes = 0;
```
Вот тут и надо использовать через переменные,  потому что при старте новой игры их нужно будет обнулять

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+    checkWordsCount();
+
+    // рендерим новое слово
+
```
Вот тут надо перезаписать word и работать с переменной, а не с spans[i].textContent (82 строка)
word = getword();
renderWord(word);

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+
+    if (event.key === "Shift" || event.key === "Control" || event.key === "CapsLock") return; // не принимать за ошибку нажатие смены языка и капслок
+
+    if (event.key.toLowerCase() === spans[i].textContent) { // если нажата верная клавиша
```
Вместо spans[i].textContent (не делают так) работаем с переменной word. Ниже написала где эта переменная перезаписывается

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+//const content = document.querySelector(".content");
```
Лена,  весь закомментированный код убираем

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+
+        clearTimeout(timerId);
+        alert("Вы проиграли");
+        location.reload(); // перезагрузка страницы для начала новой игры
```
Так не делают.  Если нужно заново стартануть игру,  то пишется функция например startNewGame
function startNewGame() {
    minutes = 0;
    seconds = 0;
    timerId = setInterval(startTimer, 1000); 
    сorrectWords.textContent = 0;
    wrongWords.textContent = 0;
}  и вызывается тут вместо этой строки

Будет на секунду задержка обнуления,  это из-за alert

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+
+        clearTimeout(timerId);
+        alert(`Победа. Ваше время: ${timer.textContent}`);
+        location.reload(); // перезагрузка страницы для начала новой игры
```
Удалить и вызвать startNewGame

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,142 @@
+// let minutes = 0;
+
+function startTimer() {
+    let [minutes, seconds] = timer.textContent.split(":").map(Number);
```
Это нужно удалить,  берем minutes и seconds из переменных

Leksa2304 committed:
hometask-11

доработка
